docs/authorization-introduction.md

```diff
@@ -0,0 +1,124 @@
+- [Private routes][] are routes that can be accessed only through a **Service Key** that is passed as a query string in the form `sig=<key>`. The **Service Key** is loaded by Udaru from the config file. These private routes provide functionality to create, update and delete policies,
+- Routes to access and manage the Organizations, Teams, Users and Policies hierarchies are exposed as [Public routes][].
+
+## Routes in stand-alone server and Hapijs plugin modes
+
+Each endpoint exposed by Udaru (when used as a server or as a hapi.js plugin) has an Action and a Resource associated with it. To perform a call to a certain endpoint the caller needs to have policies attached that allow him to do that specific action on that specific resource. This is the middleware access control level.
+
+The endpoints that have no Resources explicitly written are detecting the associated Resource from the route structure.
+
+Example for the authorize user endpoint '/authorization/access/{userId}/{action}/{resource*}':
+```javascript
+  config: {
+    plugins: {
+      auth: {
+        action: 'authorization:authn:access',
+        resource: 'authorization/access'
+      }
+    },
+  ...
+```
+This endpoint has attached to it an action of type `authorization:authn:access` and it represents an `authorization/access` resource.
+
+Example for the get user by id endpoint `/authorization/users/{id}/policies`:
+```javascript
+  plugins: {
+    auth: {
+      action: 'authorization:users:policy:add',
+      getParams: (request) => ({ userId: request.params.id })
+    }
+  },
+```
+The endpoint has attached to it an action `authorization:users:policy:add` and its resource is detected from the route structure.
+
+## Authorization for external applications
+
+After Udaru has an Organization with Teams, Users and attached Policies structure defined its `/authorization/access/{userId}/{action}/{resource*}` endpoint can be used to check if an user is authorized to perform a certain action against a resource.
+
+The external application works with Udaru by attaching to all endpoint requests an User ID. The User ID must be passed in the http headers as the `authorization` field. Internally knowing the User ID, Udaru retrieves data about the User like Organization ID to which the user belongs to, Teams to which the User belongs to and Policies attached to it or its parent hierarchy elements. The retrieved data is used at middleware level to authoriza the User to access the endpoint and then evaluate the endpoint request (to manage the authorization model or to check if an Action can be performed over a Resource).
+
+Example:
+```javascript
+curl -X GET --header 'Accept: application/json' --header 'authorization: <UserID>' 'http://localhost:8080/authorization/organizations'
+```
+
+Another header field that can be used only by Super Users is the `org` field. This is used for impersonating the organization as if that Super User belongs to that organization.
```

Comment:
`This is used to obtain a shallow impersonation. It will only substitute the organization ID so that the super admin can perform operation in any organization`